### PR TITLE
Adds logging to problem example block.

### PR DESCRIPTION
**Pull request by pmitros**
This pull request: 
1. Adds logging to the problem XBlock. The logs are a subset of what might be found in edx-platform, with a slight difference in semantics resulting from xblock-sdk's possibility of multiple checkers for a single problem. 
2. Adds a JSON stringify to the last PR which added logging code. This results in slightly nicer console output for large events. 

@macdiesel: Mind giving this a quick review? It's a tiny PR.

Also, if interested @cpennington @nedbat

### CODE REVIEW

File: sample_xblocks/basic/problem.py [hunk 1]
```diff
@@ -139,7 +155,7 @@ def student_view(self, context=None):
                         }
                     }
                     var handlerUrl = runtime.handlerUrl(element, 'check')
-                    $.post(handlerUrl, JSON.stringify(data)).success(handleCheckResults);
+                    $.post(handlerUrl, JSON.stringify(data)).success(handleCheckResults(data));
```
Comment (macdiesel) [inline sample_xblocks/basic/problem.py]: you shouldn't put the data in here, you want to verify the data that is returned from the result, not the data you are sending correct? you should only pass in the function you wish to have called upon response.

Comment (macdiesel) [inline sample_xblocks/basic/problem.py]: to clarify inside of the .success function.

File: sample_xblocks/basic/problem.py [hunk 2]
```diff
@@ -117,13 +117,29 @@ def student_view(self, context=None):
...
 
-                function handleCheckResults(results) {
-                    $.each(results.submitResults || {}, function(input, result) {
-                        callIfExists(runtime.childMap(element, input), 'handleSubmit', result);
-                    });
-                    $.each(results.checkResults || {}, function(checker, result) {
-                        callIfExists(runtime.childMap(element, checker), 'handleCheck', result);
-                    });
+                function handleCheckResults(data) {
+                    return function(results) {
```
Comment (macdiesel) [inline sample_xblocks/basic/problem.py]: Why wrap this function?  You can remove the inner return function(results) function here.

File: workbench/static/workbench/js/runtime/logger.js [hunk 3]
```diff
@@ -37,7 +40,7 @@ XBlock generated them and other XBlock-local metadata.
             sendRequest, has;
 
         sendRequest = function(data, options) {
-	    console.log(data);
+	    console.log(JSON.stringify(data));
```
Comment (macdiesel) [inline workbench/static/workbench/js/runtime/logger.js]: Are you sure you need to stringify the object first? It should already be in JSON. Are you expecting nested objects inside the data object?

**Comment by macdiesel** [comment]
I should have asked first, but I assume this is using jquery?

File: sample_xblocks/basic/problem.py [hunk 4]
```diff
@@ -117,13 +117,29 @@ def student_view(self, context=None):
...
+                        $.each(results.submitResults || {}, function(input, result) {
+                            callIfExists(runtime.childMap(element, input), 'handleSubmit', result);
+                        });
+                        answers = {};
```
Comment (stvstnfrd) [inline sample_xblocks/basic/problem.py]: These variables aren't being declared, causing them to leak into the global namespace.

```javascript
var answers = {};
var correct_map = {};
var key;
var event;
```

File: sample_xblocks/basic/problem.py (hunk 1 above)
Comment (stvstnfrd) [inline sample_xblocks/basic/problem.py]: Yes, this appears to be an unnecessary level of indirection. `success` expects to be a passed a function reference which can simply be `handleCheckResults`, without invoking it with the parens/arg and removing the `return function(results) {` line and its corresponding closing `}`.

File: workbench/static/workbench/js/runtime/logger.js (hunk 3 above)
Comment (stvstnfrd) [inline workbench/static/workbench/js/runtime/logger.js]: @macdiesel You don't _need_ to but each outputs slightly differently.

The latter prints a string, while the former prints a dictionary. The distinction here is that `console.log(JSON.stringify(data));` will condense the object/dictionary into a single line string which is output. On the other hand, the original `console.log(data);` will output the value as in "interactive" object; you can click arrows to selectively show/hide keys/values.

I find this (the interactive form, `console.log(data);`) to be particularly useful for large dictionaries with many keys/values. 

`stringify`ing can be useful to see a small number of keys/values in a condensed, one-line format.

File: sample_xblocks/basic/problem.py (hunk 1 above)
Comment (pmitros) [inline sample_xblocks/basic/problem.py]: I am trying to follow the logging conventions of edx-platform here, where the problem_check event has both the student answer and whether that answer is correct. The handleCheckResults is called with just which answers are correct, and does not know what the submission was, unless it is passed in. By using a closure, I have access to both.

File: sample_xblocks/basic/problem.py (hunk 2 above)
Comment (pmitros) [inline sample_xblocks/basic/problem.py]: See below. I am trying to make sure I have both the results (correct/incorrect) and the actual submitted value in one place.

File: sample_xblocks/basic/problem.py (hunk 4 above)
Comment (pmitros) [inline sample_xblocks/basic/problem.py]: Thank you. Fixed (will push in a moment). I broke the .js file out so I could run jshint for other places I had similar errors.

File: workbench/static/workbench/js/runtime/logger.js (hunk 3 above)
Comment (pmitros) [inline workbench/static/workbench/js/runtime/logger.js]: In this case, the interactive form was a little bit of a disaster for several reasons. The events are fairly small, but sometimes a little deep. Lots of clicks to see what's going on. In addition, it's nice to be able to cut-and-paste them, pretty print them, compare to edx-platform, paste into Python, etc. The string form lets us do that. I removed a redundant stringify (which was needed for AJAX, but not for console).

**Comment by pmitros** [comment]
@macdiesel Thank you for the code review. @stvstnfrd Thank you for jumping in on this as well. 

We are using JSQuery.

* I fixed or commented on the issues. There are no more global variables -- I feel dumb about that one. 
* I pulled out the JavaScript into its own file. There were many stupid mistakes, mostly in my code, sometimes outside, due to being in a .py file (which kills syntax highlighting, jshint, etc.)

**Comment by edx-webhook** [comment]
Thanks for the pull request, @pmitros! I've created [OSPR-2131](https://openedx.atlassian.net/browse/OSPR-2131) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Pull request closed without merging**